index.js

```diff
@@ -386,6 +391,81 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+
+    var threadRevisions = thread.related
+        .map(function(relatedMsg) {
+      
+        if (relatedMsg.value.content.type === 'post-edit' &&
+            // ^ make sure it's an edit
+            relatedMsg.value.content.revision &&
+            // ^ may be unnecessary if schema validates
+            (relatedMsg.value.content.revision === thread.key ||
+             relatedMsg.value.content.root     === thread.key)
```

Comment:
wait, what does `root` point to? The topmost post in a post-thread, or the very first version of a post?